ibm/service/cos/resource_ibm_cos_backup_policy.go

```diff
@@ -0,0 +1,153 @@
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"bucket_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Bucket Crn of the source bucket.",
+			},
+			"policy_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the backup policy to be applied on the source bucket.",
+			},
+			"target_backup_vault_crn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The CRN for a COS BackupVault.",
+			},
+			"backup_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"continuous"}),
+				Description:  "The type of backup to support.",
+			},
+			"policy_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Id of the backup policy applied on the source bucket",
+			},
+		},
+	}
+}
+
+func resourceIBMCOSBackupPolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketCRN := d.Get("bucket_crn").(string)
+	bucketName := strings.Split(bucketCRN, ":bucket:")[1]
+	policyName := d.Get("policy_name").(string)
+	targetBackupVaultCRN := d.Get("target_backup_vault_crn").(string)
+	backupType := d.Get("backup_type").(string)
+	rcClient, err := meta.(conns.ClientSession).CosConfigV1API()
+	if err != nil {
+		return diag.Errorf("Failed to create rc client %v", err)
+	}
+	createBucketBackupPolicyOptions := &rc.CreateBackupPolicyOptions{
+		Bucket:               aws.String(bucketName),
+		PolicyName:           aws.String(policyName),
+		TargetBackupVaultCrn: aws.String(targetBackupVaultCRN),
+		BackupType:           aws.String(backupType),
+	}
+	res, _, err := rcClient.CreateBackupPolicy(createBucketBackupPolicyOptions)
+	if err != nil {
+		return diag.Errorf("Failed to create the backup policy %s, %v", bucketName, err)
+	}
+	policyId := *res.PolicyID
+	policySetID := fmt.Sprintf("%s:%s:%s:target:%s", bucketName, policyId, policyName, targetBackupVaultCRN)
+	d.SetId(policySetID)
+	return resourceIBMCOSBackupPolicyRead(ctx, d, meta)
+}
+
+func resourceIBMCOSBackupPolicyRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	bucketName := parseBackupPolicyID(d.Id(), "bucketName")
+	policyId := parseBackupPolicyID(d.Id(), "policyId")
+	rcClient, err := meta.(conns.ClientSession).CosConfigV1API()
+	if err != nil {
+		return diag.Errorf("Failed to create rc client %v", err)
+	}
+	d.Set("policy_id", policyId)
+	getBucketBackupPolicyOptions := &rc.GetBackupPolicyOptions{
+		Bucket:   aws.String(bucketName),
+		PolicyID: aws.String(policyId),
+	}
+
+	res, _, err := rcClient.GetBackupPolicy(getBucketBackupPolicyOptions)
+	if err != nil {
+		return diag.Errorf("Error while reading the backup policy vault : %v", err)
```

Comment:
Handle the scenario of deleting the resource outside Terraform Context

Reply:
done